src/rust/bin/wallet_tool.rs

```diff
@@ -0,0 +1,580 @@
+    {
+        let export_path = export_path.to_string();
+        panic::set_hook(Box::new(move |panic_info| {
+            clear_and_show_cautions(&export_path);
+
+            let s = panic_info.payload().downcast_ref::<&str>().unwrap_or(&"");
+            eprintln!("\nPanic: {}\n{:?}", s, Backtrace::new());
+        }));
+    }
+
+    let res = (|| -> anyhow::Result<()> {
+        println!("The recovery phrase is:\n");
+
+        const WORDS_PER_LINE: usize = 3;
+
+        let words: Vec<_> = phrase.split(' ').collect();
+        let max_len = words.iter().map(|w| w.len()).max().unwrap_or(0);
+
+        for (i, word) in words.iter().enumerate() {
+            print!("{0:2}: {1:2$}", i + 1, word, max_len + 2);
+            if (i + 1) % WORDS_PER_LINE == 0 {
+                println!();
+            }
+        }
+        if words.len() % WORDS_PER_LINE != 0 {
+            println!();
+        }
+
+        println!(concat!(
+            "\nPlease write down this phrase on something durable that you will keep\n",
+            "in a secure location.\n",
+            "Press Enter when finished; then the phrase will disappear and you'll be\n",
+            "asked to re-enter a selection of words from it."
+        ));
+
+        let mut stdin = io::stdin();
+        let mut buf = String::new();
+        prompt(&mut stdin, &mut buf)?;
+
+        // The only reliable and portable way to make sure the recovery phrase
+        // is no longer displayed is to clear the whole terminal (including
+        // scrollback, if possible). The text is only printed if clearing fails.
+        try_to_clear(concat!(
+            "\n\n\n\n\n\n\n\n\n\n\n\n",
+            "Please adjust the terminal window so that you can't see the\n",
+            "recovery phrase above. After finishing the backup, close the\n",
+            "terminal window or clear it"
+        ));
+
+        println!("\nNow we're going to confirm that you backed up the recovery phrase.");
+
+        let mut rng = thread_rng();
+        let mut unconfirmed: Vec<usize> = (0..words.len()).collect();
+        for _ in 0..min(3, words.len()) {
+            let index: usize = rng.gen_range(0..unconfirmed.len());
+            let n = unconfirmed[index];
+            unconfirmed[index] = unconfirmed[unconfirmed.len() - 1];
+            unconfirmed.pop().expect("should be nonempty");
+
+            loop {
+                let mut buf = String::new();
+                println!("\nPlease enter the {} word:", ordinal(n + 1));
+                let line = prompt(&mut stdin, &mut buf)?;
+                if words[n] == line {
+                    break;
+                }
+                println!("That's not correct, please try again.");
+            }
+        }
+        Ok(())
+    })();
+
+    panic::set_hook(old_hook);
+    clear_and_show_cautions(export_path);
+    res?;
+
+    let mut cli_args = cli_options;
+    cli_args.extend_from_slice(&["walletconfirmbackup".to_string(), phrase.to_string()]);
```

Comment:
This exposes the phrase in the CLI arguments to the subprocess, and therefore in the process table. We need to ensure we are following best practice in our tool, namely using the `-stdin` flag and passing the phrase to the subprocess via stdin. As a side-effect, we also don't need to call `phrase.to_string()` to re-allocate the phrase (adding it to the heap again); we can just write the `&str` directly to the subprocess.
```suggestion
    cli_args.extend_from_slice(&["walletconfirmbackup".to_string(), "-stdin".to_string()]);
```